share/functions/__fish_config_interactive.fish

```diff
@@ -266,6 +266,13 @@ function __fish_config_interactive -d "Initializations that should be performed
         __update_cwd_osc # Run once because we might have already inherited a PWD from an old tab
     end
 
+    # Create empty configuration directores and files if they don't already exist
+    if not test -e $__fish_config_dir/completions/ -a -e $__fish_config_dir/conf.d/ \
+            -a -e $__fish_config_dir/functions/ -a -e $__fish_config_dir/config.fish
+        mkdir -p $__fish_config_dir/{completions, conf.d, functions}
+        touch $__fish_config_dir/config.fish
```

Comment:
I don't quite understand what you're saying, could you explain?

Reply:
An empty file would be weird - I don't know any tool that does that.

Reply:
ohhhhhh... Yes! I would love for that code snippet to be in `config.fish` by default.

I'd prefer `status is-interactive || exit` since it's only one line.

Reply:
> I'd prefer status is-interactive || exit since it's only one line.

The user is not necessarily a programmer, so they might not know this trick.

Reply:
As you wish.

Reply:
Counter argument (bikeshedding to the extreme 😂):
1. Calling basic shell syntax a "trick" is a bit much, though I get where you are coming from.
2. It's more boilerplate added to the user's `config.fish`.
3. The code to write that paragraph to a file just looks awful. Do you really want this
  ```fish
printf '%s\n' \
    "# Commands after this block will be ignored when running a script." \
    "if not status is-interactive" \
    "   exit" \
    end $__fish_config_dir/config.fish
  ```
  or this
  ```fish
    echo "# Do not run the commands below when running a script.
if not status is-interactive
    exit
end" >$__fish_config_dir/config.fish
  ```
in your codebase? 👀 😄

Reply:
I agree that some form of boilerplate is a good idea. We see many issues created because of commands running in non-interactive sessions that produce terminal output.

My preference would be for something along the lines of 

```fish
if status is-interactive
    # Commands to run in interactive sessions can go here
end
```

rather than using `exit` which disrupts the general flow of the script.

Reply:
> The code to write that paragraph to a file just looks awful.

We have plenty of code that's not pretty but does the job, and it's worth it.
What users see more important by orders of magnitude than what developers see.

BTW it is possible to align the first line:

```fish
echo "\
line1
line2"
```

> rather than using exit which disrupts the general flow of the script.

Makes total sense, I didn't realize that "early exit" is another idiom; your version is closer to natural language.